apps/src/blockly/addons/blockSvgLimit.ts

```diff
@@ -0,0 +1,144 @@
+import {BlockSvg} from 'blockly';
+
+const BUBBLE_SIZE = 18;
+const HALF_BUBBLE_SIZE = BUBBLE_SIZE / 2;
+/**
+ * Represents a bubble on a Blockly block, displaying the count of blocks remaining
+ * based on a limit initially stated in the toolbox XML.
+ */
+export default class BlockSvgLimit {
```

Comment:
nit: Would renaming this to `BlockLimitIndicator` be clearer? Since this class isn't the SVG block itself?

Reply:
It's a limit indicator _for_ an SVG block. Similar to `BlockSvgFrame` and `WorkspaceSvgFrame` in terms of naming convention. Since we wouldn't put one on an (unrendered) `Block`, could we do `BlockSvgLimitIndicator`?